serverless/src/submitSwodlrOrder/__tests__/handler.test.js

```diff
@@ -0,0 +1,550 @@
+import knex from 'knex'
+import mockKnex from 'mock-knex'
+import nock from 'nock'
+import { v4 as uuidv4 } from 'uuid'
+
+import * as getDbConnection from '../../util/database/getDbConnection'
+import * as startOrderStatusUpdateWorkflow from '../../util/startOrderStatusUpdateWorkflow'
+
+import { mockSwodlrOrder } from './mocks'
+
+import submitSwodlrOrder from '../handler'
+
+let dbTracker
+
+jest.mock('uuid')
+uuidv4.mockImplementation(() => 'mock-request-id')
+
+beforeEach(() => {
+  jest.clearAllMocks()
+
+  jest.spyOn(getDbConnection, 'getDbConnection').mockImplementationOnce(() => {
+    const dbCon = knex({
+      client: 'pg',
+      debug: false
+    })
+
+    // Mock the db connection
+    mockKnex.mock(dbCon)
+
+    return dbCon
+  })
```

Comment:
Do we need to mock this every time we start a test? Wouldn't `mockImplementaton` serve better here as this implementation is not changing. Then it can be outside of the before each and save resources when running?

Reply:
That is a good point, I think the idea was to mock it the way that the handler runs the test so each time it is going to be mocked per lambda call (when a job is picked up in the queue that is) mocking it once I think is to ensure that it is only making one database connection per `test` if it made more it would be uncaught and fail the test. To be fair this is the test apparatus on the other handlers such as `serverless/src/submitCatalogRestOrder/__tests__/handler.test.js` or `serverless/src/submitHarmonyOrder/__tests__/handler.test.js` which at least I would rather be consistent.  A good work item for innovation might be to see if we can reduce the amount of time that the unit tests take to run more generally

Reply:
I agree with Ed. We should keep this as it is (for the time being)